integration/spark/src/main/scala/org/apache/spark/sql/secondaryindex/util/SecondaryIndexUtil.scala

```diff
@@ -280,6 +286,35 @@ object SecondaryIndexUtil {
     }
   }
 
+  /**
+   * This method deletes the old index files or merge index file after data files merge
+   */
+  def deleteOldIndexOrMergeIndexFiles(factTimeStamp: Long,
+      validSegments: util.List[Segment],
+      indexCarbonTable: CarbonTable): Unit = {
+    // delete the index/merge index carbonFile of old data files
+    validSegments.asScala.foreach { segment =>
+      var indexFiles: util.Map[String, String] = null
+      // for old store scenario
+      if (segment.getSegmentFileName == null) {
```

Comment:
move line 299 to 307 as a function in CarbonTable to return the index file list for specified segment list

Reply:
done